packages/docs-page/types.ts

```diff
@@ -0,0 +1,37 @@
+interface Node {
+  /**
+   * @example
+   * ```tsx
+   * const title = "AliCloud With A Very Long NavLeaf Title That Should Wrap To Multiple Lines"
+   * const title = <code>GCP</code>
+   * ```
+   */
+  title: string | React.ReactNode
+  hidden?: boolean
+}
+
+interface NavItem extends Node {
+  /** @example `"agent/autoauth/methods/alicloud"` */
+  path: string
+}
+interface ExternalItem extends Node {
+  /**
+   * An external link, or an internal direct link
+   * @example `"https://google.com"`, `"/components/docssidenav"`
+   */
+  href: string
+}
+
+interface Divider {
+  divider: true
+}
+
+interface NavTree extends Node {
+  routes: (NavItem | ExternalItem | NavTree | Divider)[]
+}
+
+export interface NavData {
+  title: string
+  path: string
+  routes: NavTree
+}
```

Comment:
I think these types are also repeated here: https://github.com/hashicorp/mktg-content-workflows/blob/main/shared/types.ts

I wonder if it would make sense to elevate these types to `@hashicorp/platform-types`. What do you think?

Reply:
Actually, it's probably best to pull from here! https://github.com/hashicorp/react-components/blob/main/packages/docs-sidenav/types.ts

Reply:
I'll delete this in favor of `import { NavData } from '@hashicorp/react-docs-sidenav/types'`

Reply:
Addressed in https://github.com/hashicorp/react-components/pull/337/commits/a25c71a0b9b00298235785c04d4e10e095b7f63f